media_kit_video_controls/lib/src/controls/adaptive.dart

```diff
@@ -0,0 +1,27 @@
+/// This file is a part of media_kit (https://github.com/alexmercerind/media_kit).
+///
+/// Copyright © 2021 & onwards, Hitesh Kumar Saini <saini123hitesh@gmail.com>.
+/// All rights reserved.
+/// Use of this source code is governed by MIT license that can be found in the LICENSE file.
+// ignore_for_file: non_constant_identifier_names
+import 'package:flutter/material.dart';
+import 'package:media_kit_video/media_kit_video.dart';
+
+/// {@template adaptive_video_controls}
+///
+/// [Video] controls based on the running platform.
+///
+/// {@endtemplate}
+Widget AdaptiveVideoControls(VideoState state) {
```

Comment:
When `CupertinoVideoControls` will be implemented, I think it will make sense to select the right controls for `MaterialApp` and `CupertinoApp` :
```dart
final isInMaterialApp = state.context.findAncestorWidgetOfExactType<MaterialApp>() != null ;
final isInCupertinoApp = state.context.findAncestorWidgetOfExactType<CupertinoApp>() != null ;
```

Reply:
Also, selecting the layout for mobile or desktop based on size rather than platform may be a better choice.

Reply:
Currently, the reason for choosing controls based on the platform is that `MaterialVideoControls` is **"touch based"**, while `MaterialDesktopVideoControls` is **"mouse & keyboard based"** e.g.

- MaterialDesktopVideoControls: Hover to hide & show controls.
- MaterialDesktopVideoControls: Keyboard shortcuts.
- MaterialVideoControls: Tap to hide & show controls.
- MaterialVideoControls: Double tap to seek.
- MaterialVideoControls: Vertical swipe for brightness/volume adjustment.

Although both controls are usable on touch/keyboard-mouse or vice-versa (& I have used them); they certainly feel "odd" & "out-of-place".

<hr>

On the other hand, dependents can still:

- Select controls by passing the `controls` argument in `Video` widget.
- Change the default controls on each platform by writing their own method similar to `AdaptiveVideoControls`; which uses size (whatever they like) etc.
- Create custom controls from scratch.

Reply:
> When `CupertinoVideoControls` will be implemented, I think it will make sense to select the right controls for `MaterialApp` and `CupertinoApp` :
> 
> ```dart
> final isInMaterialApp = state.context.findAncestorWidgetOfExactType<MaterialApp>() != null ;
> final isInCupertinoApp = state.context.findAncestorWidgetOfExactType<CupertinoApp>() != null ;
> ```

Sure.